src/costmanagement/azext_costmanagement/manual/custom.py

```diff
@@ -0,0 +1,163 @@
+                                 schedule_status=None,
+                                 schedule_recurrence=None,
+                                 schedule_recurrence_period=None):
+
+    export_instance = client.get(scope=scope, export_name=export_name)
+
+    print(type(export_instance))
+    print(export_instance)
+
+    if isinstance(definition_dataset_aggregation, str):
+        definition_dataset_aggregation = json.loads(definition_dataset_aggregation)
+    if isinstance(definition_dataset_filter, str):
+        definition_dataset_filter = json.loads(definition_dataset_filter)
+
+    delivery_info_destination = {
+        'resource_id': delivery_storage_account_id or export_instance.destination.resource_id,
+        'container': delivery_storage_container or export_instance.destination.container,
+        'root_folder_path': delivery_directory or export_instance.destination.root_folder_path,
+    }
+
+    with cmd.update_context(export_instance) as c:
+        # update export schedule configuration
+        c.set_param('status', schedule_status)
+        c.set_param('recurrence', schedule_recurrence)
+        c.set_param('recurrence_period', schedule_recurrence_period)
+
+        # update delivery info
+        c.set_param('destination', delivery_info_destination)
+
+        # update export definition
+        c.set_param('timeframe', definition_timeframe)
+        c.set_param('configuration', definition_dataset_configuration)
+        c.set_param('aggregation', definition_dataset_aggregation)
+        c.set_param('grouping', definition_dataset_grouping)
+        c.set_param('filter', definition_dataset_filter)
+
+    print('-' * 100)
+    print(export_instance)
+
+    return client.create_or_update(scope=scope,
+                                   export_name=export_name,
+                                   type=export_instance.type_properties_definition_type,
+                                   timeframe=export_instance.timeframe,
+                                   time_period=export_instance.time_period,
+                                   configuration=export_instance.configuration,
+                                   aggregation=export_instance.aggregation,
+                                   grouping=export_instance.grouping,
+                                   filter=export_instance.filter,
+                                   destination=export_instance.destination,
+                                   status=export_instance.status,
+                                   recurrence=export_instance.recurrence,
+                                   recurrence_period=export_instance.recurrence_period,
+                                   e_tag=export_instance.e_tag)
+
+
+def costmanagement_export_list(cmd, client, scope):
+    return client.list(scope=scope).value   # value exist even the result is empty
+
+
+def costmanagement_export_show(cmd, client,
+                               scope,
```

Comment:
=None?

Reply:
This is copied from `generated/` without modification.

Reply:
The logic is a little bit strange for me. We checked the `scope` is equal to `None` or not. Should we modify it?

---
In reply to: [432214219](https://github.com/Azure/azure-cli-extensions/pull/1729#discussion_r432214219) [](ancestors = 432214219)

Reply:
Fixed